scripts/detect_duplicate_features.py

```diff
@@ -0,0 +1,84 @@
+import os
+import argparse
+
+import capa.rules
+import capa.engine as ceng
+
+
+def get_child_features(feature):
+    children = []
+
+    if isinstance(feature, (ceng.And, ceng.Or, ceng.Some)):
+        for child in feature.children:
+            children.extend(get_child_features(child))
+    elif isinstance(feature, (ceng.Subscope, ceng.Range, ceng.Not)):
+        children.extend(get_child_features(feature.child))
+    else:
+        children.append(feature)
+    return children
+
+
+def get_features(rule_path, errors):
+    with open(rule_path, "r") as f:
+        feature_list = []
+        try:
+            new_rule = capa.rules.Rule.from_yaml(f.read())
+            feature_list = get_child_features(new_rule.statement)
+        except Exception as e:
+            errors.append("rule :" + rule_path + " " + str(type(e)) + " " + str(e))
+    return feature_list, errors
+
+
+def find_overlapping_rules(new_rule_path, rules_path):
+    if not new_rule_path.endswith(".yml"):
+        raise FileNotFoundError("FileNotFoundError ! New rule file name doesn't end with yml")
+
+    new_rule_features, error = get_features(new_rule_path, [])
+    if error:
+        raise Warning(error[0])
+
+    errors: list = []
+    count = 0
+    overlapping_rules = []
+    for rules in rules_path:
```

Comment:
there's also the `get_rules` function we could use here

Reply:
Thank you for letting me know about the get_rules function. I wasn't aware of it before, I'll definitely look into it.

Reply:
> there's also the `get_rules` function we could use here

Hey @gmr-tz ,
When I ran `get_rules` on `.\rules` path I got a rules set contaning 1013 Rules.
Also when I ran `capa.main.collect_rule_file_paths` on the same path I got total 796 rule paths. 
These extra 217 rules seem to come from `capa.rules.cache.load_cached_ruleset`. Will these extra rules cause any issues.

Reply:
These extra rules come from internally generated subscope rules to facilitate matching.

Reply:
Do these rules also need to get checked for overlap?

Reply:
The only difference I found in using `get_rules` is that instead of path to rules as in my code it will be returning only the name of rules.

Reply:
@gmr-tz ,
I have rewritten the code using `get_rules.` Please look at the updated code when u have a chance.